cpp/src/neighbors/detail/reachability_types.cuh

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
```

Comment:
Not a reason to hold up this PR, but since cuVS is no longer header-only, we don't have to have these `_types` files anymore. We could just include this in the shared code (assuming there is a shared place to put it).

Reply:
Placed it back in `reachability.cuh`. 

The mutual reachability function will be cleaned up in another PR, and the `reachability.cuh` file will eventually be left with only `ReachabilityPostProcess` struct.